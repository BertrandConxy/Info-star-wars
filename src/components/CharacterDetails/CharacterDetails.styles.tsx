import styled from 'styled-components'
import { IoChevronBackCircleOutline } from 'react-icons/io5'

export const CharacterContainer = styled.div`
  border: 1px solid white;
  margin: 18px auto;
  width: 80%;
  padding: 10px;
`
export const ContentContainer = styled.div`
  width: 70%;
  margin: 40px auto;
`

export const Content = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 10px;
`

export const Bold = styled.span`
  font-weight: bold;
`

export const Name = styled.h2`
  font-weight: bold;
  text-align: center;

  @media screen and (max-width: 768px) {
    margin-left: 25px;
  }
`

export const Header = styled.div`
  display: flex;
  position: relative;
  justify-content: center;
  width: 70%;
  margin: 0 auto;
`
export const BackIcon = styled(IoChevronBackCircleOutline)`
  font-size: 32px;
  position: absolute;
  top: 0;
  left: 0;
  cursor: pointer;
`
