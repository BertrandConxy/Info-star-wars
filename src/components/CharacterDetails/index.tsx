import {
  CharacterContainer,
  Name,
  ContentContainer,
  Content,
  Bold,
  BackIcon,
  Header,
} from './CharacterDetails.styles'

import { iCharacter } from '../../typeDefs/character'

import { useNavigate } from 'react-router-dom'

export default function CharacterDetails({
  name,
  eye_color,
  hair_color,
  gender,
  birth_year,
  films,
  height,
  mass,
}: iCharacter) {
  const navigate = useNavigate()
  const handleBack = () => {
    navigate(-1)
  }
  return (
    <CharacterContainer>
      <Header>
        <BackIcon onClick={handleBack} />
        <Name>{name}</Name>
      </Header>
      <ContentContainer>
        <Content>
          <Bold>Birth Year:</Bold>
          <span>{birth_year}</span>
        </Content>
        <Content>
          <Bold>Gender:</Bold>
          <span>{gender}</span>
        </Content>
        <Content>
          <Bold>Eye Color:</Bold>
          <span>{eye_color}</span>
        </Content>
        <Content>
          <Bold>Hair Color:</Bold>
          <span>{hair_color}</span>
        </Content>
        <Content>
          <Bold>Height:</Bold>
          <span>{height}</span>
        </Content>
        <Content>
          <Bold>Mass:</Bold>
          <span>{mass}</span>
        </Content>
        <Content>
          <Bold>Films Featured:</Bold>
          <div>
            {films.map((film) => (
              <p>{film}</p>
            ))}
          </div>
        </Content>
      </ContentContainer>
    </CharacterContainer>
  )
}
