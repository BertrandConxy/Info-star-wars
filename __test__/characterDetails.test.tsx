import React from 'react'
import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import CharacterDetails from '../src/components/CharacterDetails/index'

const mockCharacter = {
  id: 1,
  name: 'Luke Skywalker',
  birth_year: '19 BBY',
  gender: 'Male',
  hair_color: 'Blond',
  eye_color: 'Blue',
  height: 172,
  mass: 77,
  films: ['A New Hope', 'The Empire Strikes Back', 'Return of the Jedi'],
}

describe('CharacterDetails', () => {
  beforeEach(() => {
    render(
      <MemoryRouter>
        <CharacterDetails character={mockCharacter} />
      </MemoryRouter>,
    )
  })

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterDetails character={mockCharacter} />
      </MemoryRouter>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the character name', () => {
    expect(screen.getByText('Luke Skywalker')).toBeInTheDocument()
  })

  it('renders the character details', () => {
    expect(screen.getByText('Birth year:')).toBeInTheDocument()
    expect(screen.getByText('19 BBY')).toBeInTheDocument()
    expect(screen.getByText('gender:')).toBeInTheDocument()
    expect(screen.getByText('Male')).toBeInTheDocument()
    expect(screen.getByText('Eye color:')).toBeInTheDocument()
    expect(screen.getByText('Blue')).toBeInTheDocument()
    expect(screen.getByText('Hair color:')).toBeInTheDocument()
    expect(screen.getByText('Blond')).toBeInTheDocument()
    expect(screen.getByText('Height:')).toBeInTheDocument()
    expect(screen.getByText('172')).toBeInTheDocument()
    expect(screen.getByText('Mass:')).toBeInTheDocument()
    expect(screen.getByText('77')).toBeInTheDocument()
    expect(screen.getByText('Films Featured:')).toBeInTheDocument()
    expect(screen.getByText('A New Hope')).toBeInTheDocument()
    expect(screen.getByText('The Empire Strikes Back')).toBeInTheDocument()
    expect(screen.getByText('Return of the Jedi')).toBeInTheDocument()
  })
})
