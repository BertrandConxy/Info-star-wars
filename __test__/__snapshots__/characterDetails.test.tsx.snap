// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetails matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB iwkgmW"
  >
    <div
      class="sc-jTrPJq hwLAJz"
    >
      <svg
        class="sc-gLDzan kAAHGI"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 64C150 64 64 150 64 256s86 192 192 192 192-86 192-192S362 64 256 64z"
          fill="none"
          stroke-miterlimit="10"
          stroke-width="32"
        />
        <path
          d="M296 352l-96-96 96-96"
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="32"
        />
      </svg>
      <h2
        class="sc-eDDNvR iGvxlD"
      >
        Luke Skywalker
      </h2>
    </div>
    <div
      class="sc-gueYoa iYeNhi"
    >
      <div
        class="sc-dmqHEX kXDsOq"
      >
        <span
          class="sc-hLseeU kvpNKp"
        >
          Birth year:
        </span>
        <span>
          19 BBY
        </span>
      </div>
      <div
        class="sc-dmqHEX kXDsOq"
      >
        <span
          class="sc-hLseeU kvpNKp"
        >
          gender:
        </span>
        <span>
          Male
        </span>
      </div>
      <div
        class="sc-dmqHEX kXDsOq"
      >
        <span
          class="sc-hLseeU kvpNKp"
        >
          Eye color:
        </span>
        <span>
          Blue
        </span>
      </div>
      <div
        class="sc-dmqHEX kXDsOq"
      >
        <span
          class="sc-hLseeU kvpNKp"
        >
          Hair color:
        </span>
        <span>
          Blond
        </span>
      </div>
      <div
        class="sc-dmqHEX kXDsOq"
      >
        <span
          class="sc-hLseeU kvpNKp"
        >
          Height:
        </span>
        <span>
          172
        </span>
      </div>
      <div
        class="sc-dmqHEX kXDsOq"
      >
        <span
          class="sc-hLseeU kvpNKp"
        >
          Mass:
        </span>
        <span>
          77
        </span>
      </div>
      <div
        class="sc-dmqHEX kXDsOq"
      >
        <span
          class="sc-hLseeU kvpNKp"
        >
          Films Featured:
        </span>
        <div>
          <p>
            A New Hope
          </p>
          <p>
            The Empire Strikes Back
          </p>
          <p>
            Return of the Jedi
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
